espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/KReflectionUtils.kt

```diff
@@ -42,6 +43,29 @@ object KReflectionUtils {
                     } catch (e:ClassCastException) {
                         // Ignore class cast exceptions and don't try matching String to Enum
                     }
+
+                    // Handle the Class case
+                    val classifier = type.classifier;
+                    if (classifier is KClass<*> && classifier.isSubclassOf(Class::class)) {
+                        var className: String = providedParam.toString()
+                        if (className.endsWith(".class")) {
+                            className = className.substring(0, className.length - ".class".length)
```

Comment:
takeLast

Reply:
`takeLast` keeps the last characters

I changed it it to `className.take(className.length - ".class".length)`